pkg/controller/operconfig/configmap_controller.go

```diff
@@ -0,0 +1,78 @@
+package operconfig
+
+import (
+	"bytes"
+	"context"
+	"log"
+
+	"github.com/openshift/cluster-network-operator/pkg/names"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// newConfigMapReconciler returns a new reconcile.Reconciler
+func newConfigMapReconciler(client client.Client) *ReconcileConfigMaps {
+	return &ReconcileConfigMaps{client: client}
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMaps{}
+
+// ReconcileConfigMaps watches for updates to specified resources
+type ReconcileConfigMaps struct {
+	client client.Client
+}
+
+// Reconcile updates ValidatingWebhookConfiguration CABundle, given from SERVICE_CA_CONFIGMAP
+func (r *ReconcileConfigMaps) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace != names.APPLIED_NAMESPACE || request.Name != names.SERVICE_CA_CONFIGMAP {
+		return reconcile.Result{}, nil
+	}
+	log.Printf("Reconciling update to %s/%s\n", request.Namespace, request.Name)
+
+	caBundleConfigMap := &corev1.ConfigMap{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, caBundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+
+	caBundleData, ok := caBundleConfigMap.Data["service-ca.crt"]
+	if !ok {
+		// Need to wait to fill the service-ca.crt by ca-operator.
+		return reconcile.Result{}, err
+	}
+
+	webhookConfig := &admissionregistrationv1beta1.ValidatingWebhookConfiguration{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: names.MULTUS_VALIDATING_WEBHOOK}, webhookConfig)
+	if errors.IsNotFound(err) {
+		log.Println("Webhook is not found.")
+		return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+	}
+
+	webhookConfig = webhookConfig.DeepCopy()
+	modified := false
+	for idx, webhook := range webhookConfig.Webhooks {
+		// Update CABundle if CABundle is empty or updated.
+		if webhook.ClientConfig.CABundle == nil || !bytes.Equal(webhook.ClientConfig.CABundle, []byte(caBundleData)) {
+			modified = true
+			webhookConfig.Webhooks[idx].ClientConfig.CABundle = []byte(caBundleData)
+		}
+	}
+	if !modified {
+		return reconcile.Result{}, err
+	}
+
+	// Update webhookConfig
+	err = r.client.Update(context.TODO(), webhookConfig)
```

Comment:
If the webhook configuration is created by the render phase, then won't it be overridden by the next render phase? You may need to update the reconciliation logic, or create the object here.

There is probably a best practice for this which I am not aware of.

Reply:
In this case, we need to bring only BundleCA value ConfigMap value to webhook. So ConfigMap value is not changed and webhook is already created initially, so it should make sense.

Could you please let me know the reason why need to create WebhookConfig here?
Do you mean we need to update ConfigMap at that time? If so what should be changed in ConfigMap?

Reply:
I've check WebhookConfigration with `oc get ValidatingWebhookConfiguration multus.openshift.io -o jsonpath='{.webhooks[0].clientConfig.caBundle}' --watch=true`. I can see same output (same caBundle) periodically (about 5min) without any changes. So CNO touch the webhook but caBundle is not changed.